### PR TITLE
[pylint] re-enable no-else-return and inconsistent-return-statements

### PR DESCRIPTION
**Pull request by pponnuvel**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [x] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

File: sos/collector/__init__.py [hunk 1]
```diff
@@ -619,18 +619,13 @@ def _validate_option(self, default, cli):
...
-                else:
-                    return False
-        self.exit(f"Unknown option type: {cli.opt_type}")
+
+        val = cli.value.lower()
+        if val not in ['true', 'on', 'yes', 'false', 'off', 'no']:
+            msg = ("Invalid value for %s. Accepted values are: 'true', "
+                   "'false', 'on', 'off', 'yes', 'no'.")
+            self.exit(msg % cli.name, 1)
+        return val in ['true', 'on', 'yes']
```
Comment (TurboTurtle) [inline sos/collector/__init__.py:628]: We need to account for non-matching/indeterminate option types here.

**Review by TurboTurtle — COMMENTED** [review]
Looks mostly good to me, just the one note.

File: sos/collector/__init__.py (hunk 1 above)
Comment (TurboTurtle) [reply sos/collector/__init__.py:628]: Actually, looking at this again, perhaps the unknown type is superfluous.

Comment (pponnuvel) [reply sos/collector/__init__.py:628]: yeah, the unknown value is already covered by the previous `if` clause. So the final `self.exit(f"Unknown option type: {cli.opt_type}")` was always an unreachable code.

**Comment by pponnuvel** [comment]
Also addressed `inconsistent-return-statements` here (combined https://github.com/sosreport/sos/pull/3726 with this one).

File: sos/policies/distros/redhat.py [hunk 2]
```diff
@@ -138,9 +138,8 @@ def check_usrmove(self, pkgs):
         """
         if 'filesystem' not in pkgs:
             return os.path.islink('/bin') and os.path.islink('/sbin')
-        else:
-            filesys_version = pkgs['filesystem']['version']
-            return True if filesys_version[0] == '3' else False
+        filesys_version = pkgs['filesystem']['version']
+        return True if filesys_version[0] == '3' else False
```
Comment (pmoravec) [inline sos/policies/distros/redhat.py:142]: Isn't `return (filesys_version[0] == '3')` more concise and readable / understandable?

Anyway, that is a change different than fixinf the pylint issue, ACK to the current change either way.

**Review by pmoravec — APPROVED** [review]

File: sos/policies/distros/redhat.py (hunk 2 above)
Comment (pponnuvel) [reply sos/policies/distros/redhat.py]: yeah, fixed that now.

**Review by pmoravec — APPROVED** [review]

**Review by TurboTurtle — APPROVED** [review]

**Pull request merged**